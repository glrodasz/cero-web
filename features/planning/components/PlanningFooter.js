import PropTypes from 'prop-types'
import { Spacer, Paragraph, Button } from '@glrodasz/components'

const PlanningFooter = ({ tasksLength, onClickStartSession }) => {
  if (!!tasksLength >= 1) {
    return (
      <>
        <Spacer.Horizontal size="lg" />
        <Paragraph size="sm">
          Basados en la matriz de Eisenhower priorizamos tus tareas evitando
          listas de pendientes saturadas.
        </Paragraph>
        <Spacer.Horizontal size="sm" />
        <Button onClick={onClickStartSession} isDisabled type="primary">
          Empieza ahora
        </Button>
      </>
    )
  }

  return null
}

PlanningFooter.propTypes = {
  onClickStartSession: PropTypes.func.isRequired,
  tasksLength: PropTypes.number,
}

export default PlanningFooter
