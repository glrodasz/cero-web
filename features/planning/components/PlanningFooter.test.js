import { render, screen, fireEvent } from '@testing-library/react'
import PlanningFooter from './PlanningFooter'

jest.mock('@glrodasz/components', () => {
  const { dummyRender } = require('../../../utils/testUtils/dummyRender')
  const originalModule = jest.requireActual('@glrodasz/components')
  return {
    ...originalModule,
    Spacer: { Vertical: dummyRender('Spacer.Vertical') },
    Paragraph: dummyRender('Paragraph'),
  }
})

describe('[ features / planning / components / PlanningFooter ]', () => {
  describe('when `tasksLength` is greater or equal to one', () => {
    it('should return the footer', () => {
      // Arrange
      const props = {
        tasksLength: 1,
        onClickStartSession: () => {},
      }

      // Act
      const { asFragment } = render(<PlanningFooter {...props} />)
      const result = asFragment()

      // Assert
      expect(result).toMatchSnapshot()
    })
  })

  describe('when `tasksLength` is less than one', () => {
    it('should return null', () => {
      // Arrange
      const props = {
        tasksLength: 0,
        onClickStartSession: () => {},
      }
      const expected = null

      // Act
      const { container } = render(<PlanningFooter {...props} />)
      const result = container.firstChild

      // Assert
      expect(result).toBe(expected)
    })
  })

  describe('when the `Button` is clicked', () => {
    it('should call `onClickStartSession`', () => {
      // Arrange
      const onClickStartSessionStub = jest.fn()
      const props = {
        tasksLength: 1,
        onClickStartSession: onClickStartSessionStub,
      }

      // Act
      render(<PlanningFooter {...props} />)
      fireEvent.click(screen.getByText('Empieza ahora'))

      // Assert
      expect(onClickStartSessionStub).toHaveBeenCalled()
    })
  })
})
