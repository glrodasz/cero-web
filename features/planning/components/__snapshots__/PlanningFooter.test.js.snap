// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ features / planning / components / PlanningFooter ] when \`tasksLength\` is greater or equal to one should return the footer 1`] = `
<DocumentFragment>
  
[Spacer.Vertical {"size":"lg"} /]
[Paragraph {"size":"sm"}]Basados en la matriz de Eisenhower priorizamos tus tareas evitando listas de pendientes saturadas.[/Paragraph]
[Spacer.Vertical {"size":"sm"} /]
  <button
    class="button type-primary"
  >
    <h1
      class="heading color-inverted size-md weight-bold is-centered"
    >
      Empieza ahora
    </h1>
  </button>
</DocumentFragment>
`;
